pkg/nspawntool/kubeadm.go

```diff
@@ -88,7 +89,18 @@ func JoinNode(cfg *config.ClusterConfiguration, mNo int) error {
 	}
 	joinCmd = append(joinCmd, []string{
 		"/usr/bin/kubeadm", "join", "--skip-preflight-checks",
-		"--token", cfg.Token,
-		cfg.Machines[0].IP + ":6443"}...)
+		"--token", cfg.Token}...)
+
+	// --discovery-token-unsafe-skip-ca-verification appeared in Kubernetes 1.8
+	// See: https://github.com/kubernetes/kubernetes/pull/49520
+	// It is mandatory since Kubernetes 1.9
+	// See: https://github.com/kubernetes/kubernetes/pull/55468
+	// Test is !<1.8 instead of >=1.8 in order to handle non-semver version 'latest'
+	if !utils.CheckVersionConstraint(cfg.KubernetesVersion, "<1.8") {
```

Comment:
FYI, similar version checks are done in `pkg/config/defaults.go`, either `if cfg.DevCluster || utils.CheckVersionConstraint(cfg.KubernetesVersion, ">=1.8.0")` or `if !cfg.DevCluster && utils.CheckVersionConstraint(cfg.KubernetesVersion, "<1.8.0")`. In this file the check is now used a little differently. Basically this pattern is kind of error-prone, so I think we should be somehow consistent. We could make a helper to simplify it in another PR.